deepchecks/nlp/text_data.py

```diff
@@ -392,50 +294,24 @@ def calculate_default_properties(self, include_properties: t.List[str] = None,
         properties, properties_types = calculate_default_properties(
             self.text, include_properties=include_properties, ignore_properties=ignore_properties,
             include_long_calculation_properties=include_long_calculation_properties, device=device)
-        self._properties = pd.DataFrame(properties, index=self.index)
-        self._properties_types = properties_types
-
-    def set_properties(self, properties: pd.DataFrame, properties_types: t.Optional[t.Dict[str, str]] = None):
-        """Set the properties of the dataset."""
-        if self._properties is not None:
-            warnings.warn('Properties already exist, overwriting them', UserWarning)
-
-        if not isinstance(properties, pd.DataFrame):
-            raise DeepchecksValueError(f'properties type {type(properties)} is not supported, must be a'
-                                       f' pandas DataFrame')
-        if list(properties.index) != self.index:
-            raise DeepchecksValueError('properties index must be the same as the text data index')
-        self._properties = properties
-
-        if properties_types is None:
-            # TODO: move infer_categorical_features to core
-            cat_features = infer_categorical_features(properties)
-            properties_types = {
-                properties.columns[i]: 'categorical' if properties.columns[i] in cat_features else 'numeric'
-                for i in range(len(properties.columns))}
-        elif sorted(list(properties_types.keys())) != sorted(list(properties.columns)):
-            raise DeepchecksValueError('properties_types keys must identical to properties columns')
-
+        self._properties = pd.DataFrame(properties, index=self.get_original_text_indexes())
         self._properties_types = properties_types
 
     @property
     def properties(self) -> pd.DataFrame:
         """Return the properties of the dataset."""
         if self._properties is None:
-            raise DeepchecksNotSupportedError(
-                'TextData does not contain properties, add them by using calculate_default_properties or '
-                'set_properties functions')
+            get_logger().info('No properties are set, calculating default properties')
+            self.calculate_default_properties()
```

Comment:
Not wanted behavior, we discussed this in the past. As properties can take a long time to calculate, and they are not saved to the TextData object (as we take a copy of the user's original object and not the original one), then calculating them in the Suite/Checks will result in long running times + will have to be run again anyway.

So we decided that properties should be precalculated...

Reply:
Ok, reverted to previous